src_code/src/controllers/services/Authentication/SignUp.js

```diff
@@ -0,0 +1,163 @@
+/** Initialize neccessary modules */
+const bcrypt = require("bcryptjs");
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+const dbHelper = require("../../../utils/interfaces/IDBHelperFunctions");
+
+/**
+ * Handle user sign-up and insert the new account. The password is encrypted.
+ *
+ * @param {object} res - The response object for the HTTP request.
+ * @param {object} req - The request object from the HTTP request.
+ * @returns {object} - This response object indicates the result of the sign-up attempt.
+ * 
+ * Expected Request Body: 
+ * {
+ *      "userRole": "Student",
+ *      "firstName": "Le",
+ *      "middleName": "",
+ *      "lastName": "Tran",
+ *      "studentId": "", 
+ *      "emailAddress": "tran12312@spu.edu",
+ *      "accountPassword":"TallasGiraffe1!"
+ * }
+ */
+async function SignUp(res, req) {
+    try{ 
+        /** Validate if Sign Up information from request body is valid */
+        const errors = await SignUpValidation(res, req);
+        if (errors) {
+            return errors;
+        } 
+        
+        /** Destructure variables from the request body */
+        const {userRole, firstName, middleName, lastName, emailAddress, accountPassword } = req;
+        
+        /** Handle if emtpy studentId, it should be null instead of empty string */
+        if(req.studentId.length === 0 || req.studentId === "") {
+            req.studentId = null;
+        } else {
+            /** If there is studentId, trim trailing leading spaces */
+            req.studentId = req.studentId.trim();
+        }
+
+        /** Generate salt for encrypting password */
+        const salt = await bcrypt.genSalt(10); 
+
+        /** Encrypt the password */
+        const encryptedPassword = await bcrypt.hash(accountPassword, salt);
+
+        /** Prepare data we insert to the table */
+        const insertData = {
+            USER_ROLE: userRole.trim(),
+            FIRST_NAME: firstName.trim(),
+            MIDDLE_NAME: middleName.trim(),
+            LAST_NAME: lastName.trim(),
+            STUDENT_ID: req.studentId,
+            EMAIL_ADDRESS: emailAddress.trim(),
+            ACCOUNT_PASSWORD: encryptedPassword,
+        }
+
+        /** Inserting 'insertData' into the table */
+        await db("user_info").insert(insertData);
+
+        /** Return 'Sign up successfully' message after insert to table successfully */
+        return responseBuilder.BuildResponse(res, 200, {
+            message: "Sign up successfully."
+        });
+    } catch(error){ 
+        /** If an unexpected server error occurs while signing up after passing validation, return log and error. */
+        console.log("ERROR: An error occurred while signing up: ", error);
```

Comment:
Yet another console.log for testing?

Reply:
It is for AWS Cloudwatch Log to log bug, that will not be appeared on the frontend.